raiden/network/transport/matrix/transport.py

```diff
@@ -955,79 +966,75 @@ def _get_room_for_address(
         address_hex = to_normalized_address(address)
         msg = f"address not health checked: me: {self._user_id}, peer: {address_hex}"
         assert address and self._address_mgr.is_address_known(address), msg
+        assert self._raiden_service is not None
 
-        # filter_private is done in _get_room_ids_for_address
-        room_ids = self._get_room_ids_for_address(address)
-        if room_ids:  # if we know any room for this user, use the first one
-            # This loop is used to ignore any broadcast rooms that may have 'polluted' the
-            # user's room cache due to bug #3765
-            # Can be removed after the next upgrade that switches to a new TokenNetworkRegistry
-            while room_ids:
-                room_id = room_ids.pop(0)
-                room = self._client.rooms[room_id]
-                if not self._is_broadcast_room(room):
-                    self.log.debug("Existing room", room=room, members=room.get_joined_members())
-                    return room
-                self.log.warning("Ignoring broadcast room for peer", room=room, peer=address_hex)
-
-        assert self._raiden_service is not None, "_raiden_service not set"
-        address_pair = sorted(
-            [to_normalized_address(address) for address in [address, self._raiden_service.address]]
-        )
-        room_name = make_room_alias(self.chain_id, *address_pair)
-
-        # no room with expected name => create one and invite peer
+        # filter peer_candidates
         peer_candidates = [
             self._get_user(user) for user in self._client.search_user_directory(address_hex)
         ]
-
-        # filter peer_candidates
         peers = [user for user in peer_candidates if validate_userid_signature(user) == address]
```

Comment:
This should at most be a list of one element, no?

Reply:
It can have as many entries as we have transport servers in the worst case.

Reply:
> It can have as many entries as we have transport servers in the worst case.

please document with a comment in the code